### ISSUE TITLE
Simple LSTM Cell interface

### ISSUE THREAD

**Issue by pcyin** [clab/dynet#949]
I am adding a simple `LSTMCell` to dynet. It is a stateless LSTM unit that takes as input the current input `x_t` and previous state `(c_tm1, h_tm1)` and outputs `(c_t, h_t)`. This is similar to `nn.LSTMCell` in pytorch, and is useful in cases like beam search, where you need to dynamically keep the set of previous hidden states.

Details can be found [here](https://github.com/pcyin/dynet/compare/master...rnn_cell). I can do `RNNCell` and `GRUCell` as well is this is something useful to include in the official repo.

**Comment by pmichel31415** [comment]
Can't you do that with RNNState in python?

**Comment by neubig** [comment]
Just my two cents: The disconnect between the C++ and Python interfaces (and the strangeness of the C++ interface) has been bothering me for a while. I think one option would be to create a unified interface where we have a number of "RNNCell", "LSTMCell", etc. classes that power all of our builders. Then you could call these in a stateless fashion, or have something like the current RNNState.

**Comment by pcyin** [comment]
Of course! However, sometimes RNNBuilder and RNNState are just too much for simple use cases. Using RNNState we need to call `.initial_state` to initialize with previous states (but right now we cannot call `.initial_state` multiple times in one cg), or call `.set_s()` to get a new state `s` with provided previous states, and then `s.add_input()`
